core/transaction.go

```diff
@@ -229,102 +231,131 @@ func (l *L1HandlerTransaction) signature() []*felt.Felt {
+		return nil, nil
 	case *L1HandlerTransaction:
-		return l1HandlerTransactionHash(t, network)
+		return l1HandlerTransactionHash(t, n)
 	case *DeployAccountTransaction:
-		return deployAccountTransactionHash(t, network)
+		return deployAccountTransactionHash(t, n)
 	default:
 		return nil, ErrUnknownTransaction
 	}
 }
 
-func deployTransactionHash(d *DeployTransaction, network utils.Network) (*felt.Felt, error) {
-	snKeccakConstructor, err := crypto.StarknetKeccak([]byte("constructor"))
-	if err != nil {
-		return nil, err
-	}
-	if d.Version.IsZero() || d.Version.IsOne() {
-		return crypto.PedersenArray(
-			new(felt.Felt).SetBytes([]byte("deploy")),
-			d.Version,
-			d.ContractAddress,
-			snKeccakConstructor,
-			crypto.PedersenArray(d.ConstructorCallData...),
-			new(felt.Felt),
-			network.ChainId(),
-		), nil
-	}
-	return nil, ErrInvalidTransactionVersion{d, d.Version.Text(10)}
-}
-
-func invokeTransactionHash(i *InvokeTransaction, network utils.Network) (*felt.Felt, error) {
-	invokeFelt := new(felt.Felt).SetBytes([]byte("invoke"))
+func invokeTransactionHash(i *InvokeTransaction, n utils.Network) (*felt.Felt, error) {
 	if i.Version.IsZero() {
-		return crypto.PedersenArray(
-			invokeFelt,
-			i.ContractAddress,
-			i.EntryPointSelector,
-			crypto.PedersenArray(i.CallData...),
-			network.ChainId(),
-		), nil
+		// Due to inconsistencies in version 0 hash calculation we don't verify the hash
+		return nil, nil
 	} else if i.Version.IsOne() {
 		return crypto.PedersenArray(
-			invokeFelt,
+			new(felt.Felt).SetBytes([]byte("invoke")),
 			i.Version,
 			i.ContractAddress,
 			new(felt.Felt),
 			crypto.PedersenArray(i.CallData...),
 			i.MaxFee,
-			network.ChainId(),
+			n.ChainId(),
 			i.Nonce,
 		), nil
 	}
 	return nil, ErrInvalidTransactionVersion{i, i.Version.Text(10)}
 }
 
-func declareTransactionHash(d *DeclareTransaction, network utils.Network) (*felt.Felt, error) {
-	declareFelt := new(felt.Felt).SetBytes([]byte("declare"))
+func declareTransactionHash(d *DeclareTransaction, n utils.Network) (*felt.Felt, error) {
 	if d.Version.IsZero() {
-		return crypto.PedersenArray(
-			declareFelt,
-			d.Version,
-			d.SenderAddress,
-			new(felt.Felt),
-			crypto.PedersenArray(make([]*felt.Felt, 0)...),
-			d.MaxFee,
-			network.ChainId(),
-			d.ClassHash,
-		), nil
+		// Due to inconsistencies in version 0 hash calculation we don't verify the hash
+		return nil, nil
 	} else if d.Version.IsOne() {
 		return crypto.PedersenArray(
-			declareFelt,
+			new(felt.Felt).SetBytes([]byte("declare")),
 			d.Version,
 			d.SenderAddress,
 			new(felt.Felt),
 			crypto.PedersenArray(d.ClassHash),
 			d.MaxFee,
-			network.ChainId(),
+			n.ChainId(),
 			d.Nonce,
 		), nil
 	}
 	return nil, ErrInvalidTransactionVersion{d, d.Version.Text(10)}
 }
 
-func l1HandlerTransactionHash(l *L1HandlerTransaction, network utils.Network) (*felt.Felt, error) {
+var unverifiableL1HandlerTxM map[felt.Felt]struct{}
+
+func populateUnverifiableL1HandlerTx() {
```

Comment:
It sucks that we have to do this:( Feels like a disgusting hack :D

Reply:
Yeah, i am just dreading populating it. We have to sit there and watch for which hash it fails and then ignore it.

Reply:
I am going to remove this piece of code since we will log for inconsistencies